update ERF interface for snow and wetland noahmp physics update

cenlinhe opened the pull request:
This PR is to update the ERF/NoahMP driver interface to accommodate the recent NoahMP source code physics updates for snowpack and wetland processes.
The NoahMP source code updates can be found from these PRs:
https://github.com/NCAR/noahmp/pull/187
https://github.com/NCAR/noahmp/pull/186
https://github.com/NCAR/noahmp/pull/175
https://github.com/NCAR/noahmp/pull/174

cenlinhe commented:
@akashdhruv Could you please do a ERF/NoahMP test for this updated version to ensure the updated driver interface works successfully?

akashdhruv commented:
@cenlinhe Yes will do.

cenlinhe commented:
> [akashdhruv](/akashdhruv)

Hi @akashdhruv If you could test this at your earliest convenience, that would be great. We need to merge this PR asap to prepare for the Noah-MP release. Thanks!

akashdhruv commented:
@cenlinhe I will finish this by Tuesday.

akashdhruv commented:
@cenlinhe I was able to review today and here are my findings. 

- Compilation error in `src/SnowRadiationSnicarMod.F90` line 568. Changing to `if (FlagSnicarUseAerosol .eqv. .true.) then` fixed it. 

- Compilation error in `src/SnowRadiationSnicarMod.F90` line 571. Changing to `if (FlagSnicarUseOC .eqv. .true.) then` fixed it. 

- Compilation error in `src/SnowRadiationSnicarMod.F90` line 1522. Changing to `write (*,"(a,e13.6,i0,i0,i0,i0)") "SNICAR ERROR: negative absoption : ", &` fixed it. 

- Segfault in `src/SurfaceAlbedoGlacierMod.F90 ` line 59 and 68. Commenting them out fixed it. 

- Segfault in `src/SurfaceAlbedoMod.F90` line 109 and 110. Commenting them out fixed it.

cenlinhe commented:
@akashdhruv Thank you for testing. 
1. I assume that the compilation error is because you need to add the two new flag variables on your ERF model code side, right?
2. What are the segfault messages? Is it related to ERF specific setup?

akashdhruv commented:
> 1. I assume that the compilation error is because you need to add the two new flag variables on your ERF model code side, right?
I don't think that is the case. 

> 2. What are the segfault messages? Is it related to ERF specific setup?
The segfault creates a Backtrace which I was able to trace it back to these lines. I suspect that I am not supplying certain variables in my namelist file. Right now I set `NSNOW=4` in my namelist file. Could that be causing an issue?

akashdhruv commented:
If you can fix the compilation based on the suggestions that worked for me, then I think this is ok. I can review the segfaults carefully and fix them on erf side.

cenlinhe commented:
Regarding the compilation, please ignore my message. I figured it out.
Regarding the segfault messages, yes, the NSNOW=4 might be the issue. In our Noah-MP model, currently we can only handle NSNOW = 3.

cenlinhe commented:
> If you can fix the compilation errors based on the suggestions that worked for me, then I think this is ok. I can review the segfaults carefully and fix them on erf side.

Yes, I will fix the compilation errors based on your suggestions. Thanks!

akashdhruv commented:
I fixed the segfaults by changing:

`SNOW_ALBEDO_OPTION  = 2  --> SNOW_ALBEDO_OPTION  = 3`

cenlinhe commented:
@RAbolafiaRosenzweig @tslin2 I have fixed all the bugs reported by @akashdhruv above in the latest commit. Please review and approve this PR at your earliest convenience.
@akashdhruv I have fixed the bug that causes the segfault error. Now you should still be able to use SNOW_ALBEDO_OPTION  = 2 in your erf namelist. The reason for the error is that those two new variables are only defined when SNOW_ALBEDO_OPTION  = 3. Now I moved their initialization to within the SNOW_ALBEDO_OPTION  = 3 if-statement.
